pages/api/games/index.js

```diff
@@ -1,11 +1,23 @@
 import dbConnect from "@/db/connect";
 import Game from "@/db/models/Game";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
+
 export default async function handler(request, response) {
   await dbConnect();
+
+  const session = await getServerSession(request, response, authOptions);
+  
   if (request.method === "GET") {
     const games = await Game.find();
     return response.status(200).json(games);
   }
+
+  if (!session) {
+    response.status(401).json({ status: "Not authorized!" });
+    return;
+  }
+
```

Comment:
inside this POST route you can get the user info by using`session.user` here directly. then you can add the information to the content received from the frontend: 

```js

 if (request.method === "POST") {
  const user = session.user
  const formData = request.body
  const newGame = {
   ...formData,
   user
   }
   Game.create(newGame)
}
```